src/hypothesis/internal/compat.py

```diff
@@ -290,16 +290,8 @@ def inner(*args, **kwargs):
         getfullargspec = silence_warnings(getfullargspec)
 
 
-def get_type_hints(thing):
-    """Try to return any type hints for ``thing``."""
-    try:
-        import typing
-        return typing.get_type_hints(thing)
```

Comment:
I'm reluctant to jump right in to using the fallback option - `get_type_hints` also resolves forward references and merges annotations from parent classes.  Since the problem seems to be that the backport can return None, why not just 

    return typing.get_type_hints(thing) or {}

to replace None with an empty dict?